### PR TITLE
Fixes a few linux issues with the makefile

### PR DESCRIPTION
**Pull request by mcdan**
Addresses #75

### CODE REVIEW

**Comment by mcdan** [comment]
This depends on PR #74 and #79 and should fix the build.

File: docker-compose/Makefile [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+
 # detect local ip of host as this is needed within containers to find the openwhisk API container
-LOCAL_IP ?= $(shell ifconfig | grep "inet " | grep -v 127.0.0.1 | cut -d\  -f2 | head -1)
-# if no IP was found, fallback to "localhost"
-ifeq ($(LOCAL_IP), )
-	LOCAL_IP = "localhost"
+ifeq ("$(UNAME_STR)","Linux")
+	LOCAL_IP=$(shell route | grep default | tr -s " " | cut -d " " -f 8 | xargs ifconfig | grep "inet addr:" | cut -d ":" -f 2 | cut -d " " -f 1)
+else
+	LOCAL_IP ?= $(shell ifconfig | grep "inet " | grep -v 127.0.0.1 | cut -d\  -f2 | head -1)
```
Comment (mcdan) [inline docker-compose/Makefile]: On my Ubuntu machine the above command produces an empty string as the result. "cut -d\  -f2 " returning a list of empty strings. The input to that pipe on my machine is:

```bash
          inet addr:172.18.0.1  Bcast:0.0.0.0  Mask:255.255.0.0
          inet addr:172.17.0.1  Bcast:0.0.0.0  Mask:255.255.0.0
          inet addr:10.32.65.11  Bcast:10.32.65.255  Mask:255.255.254.0
```

So no matter how you much the first line here you're going to get the wrong adapter.

Comment (ddragosd) [reply docker-compose/Makefile]: 👍

File: docker-compose/Makefile [hunk 2]
```diff
@@ -163,7 +173,11 @@ destroy: stop rm
 	docker ps | grep whisk | awk '{print $$1}' | xargs docker stop | xargs docker rm
 	echo "cleaning dangling docker volumes ... "
 	docker volume ls -qf dangling=true | xargs docker volume rm
-	rm -rf ~/tmp/openwhisk
+	if [ "$(UNAME_STR)" = "Linux" ]; then \
+	  sudo rm -rf ~/tmp/openwhisk ;\
```
Comment (ddragosd) [inline docker-compose/Makefile]: is the `sudo` here for Travis ? Else I'm reading that the current use can't delete things in its own home folder ?

File: docker-compose/Makefile [hunk 3]
```diff
@@ -14,12 +20,13 @@ WSK_CLI ?= $(PROJECT_HOME)/bin/wsk
...
 
 DOCKER_KERNEL ?= $(shell docker version --format "{{.Server.KernelVersion}}")
-RUNC_BINARY   ?= $(shell if [[ $(DOCKER_KERNEL) == *-moby || $(DOCKER_KERNEL) ==  *-boot2docker ]]; then (docker run --rm --privileged --pid=host debian nsenter -t 1 -m -u -n -i sh -c "which runc || which docker-runc"); else (which runc || which docker-runc); fi)
-DOCKER_BINARY ?= $(shell if [[ $(DOCKER_KERNEL) == *-moby || $(DOCKER_KERNEL) ==  *-boot2docker ]]; then (docker run --rm --privileged --pid=host debian nsenter -t 1 -m -u -n -i sh -c "which docker"); else (which docker); fi)
-
-
-UNAME_STR ?= $(shell uname)
-
+ifeq ("$(UNAME_STR)","Linux")
+	RUNC_BINARY   ?= $(shell (which runc || which docker-runc))
```
Comment (ddragosd) [inline docker-compose/Makefile]: Was this command failing  in `Linux` ? I think this happens anyway when the `DOCKER_KERNEL` is not `*-moby` or `*_boot2docker`.

Comment (mcdan) [reply docker-compose/Makefile]: So I think the default shell my make is using is bourne not bash. so the '[[' blows up straight out of the gate. I basically end around all the other stuff cause you can and should just run docker natively on linux. You can see travis failing with the same errors I was:

```bash
/bin/sh: 1: [[: not found
/bin/sh: 1: 4.4.0-101-generic: not found
/bin/sh: 1: [[: not found
/bin/sh: 1: 4.4.0-101-generic: not found
```

File: docker-compose/Makefile (hunk 2 above)
Comment (mcdan) [reply docker-compose/Makefile]: They can sure but in this case docker is writing data there. Docker is running as root on linux, so when it writes the data files for couch here, the user can't actually delete them. Detecting the current user (uid), getting it created in then injecting that into the running container is a total pain.

**Comment by ddragosd** [comment]
@mcdan I tried to solve the conflict with the `master` branch and keep the changes in this branch but it still asks to _resolve conflict_ . Mind merging in your branch and push again ?

File: docker-compose/Makefile (hunk 2 above)
Comment (mcdan) [reply docker-compose/Makefile]: I'm happy to change the Couch stuff to use a docker volume which will relieve that issue but I didn't want to make such a "big" change.

Comment (ddragosd) [reply docker-compose/Makefile]: thanks for explaining. I also prefer the `sudo` approach in this case.

File: docker-compose/Makefile (hunk 3 above)
Comment (ddragosd) [reply docker-compose/Makefile]: got it. thanks for the details @mcdan

**Comment by ddragosd** [comment]
@mcdan I'm waiting for the travis build to complete and then I'll merge this. thanks a lot for taking the time to debug these issues in order to fix the build

**Comment by mcdan** [comment]
@ddragosd - I was able to resolve with the handy github tool. didn't know I could do that.

**Pull request merged**